File: workspaces/DotNet/server/oauth.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

public class OAuth
{
    public static async Task<string> GetAccessTokenAsync()
    {
        string payPalApiBase = Config.PAYPAL_API_BASE;
        string clientId = Config.CLIENT_ID;
        string appSecret = Config.APP_SECRET;
        Console.WriteLine("payPalApiBase: " + payPalApiBase);
        Console.WriteLine("clientId: " + clientId);
        Console.WriteLine("appSecret: " + appSecret);

        string credentials = Convert.ToBase64String(Encoding.UTF8.GetBytes($"{clientId}:{appSecret}"));

        using var httpClient = new HttpClient();

        httpClient.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
        httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", credentials);

        var content = new StringContent("grant_type=client_credentials", Encoding.UTF8, "application/x-www-form-urlencoded");

        var response = await httpClient.PostAsync($"{payPalApiBase}/v1/oauth2/token", content);
        string responseContent = await response.Content.ReadAsStringAsync();
        Console.WriteLine("responseContent: " + responseContent);
        return responseContent;
    }
}

File: workspaces/PHP/server/oauth.php
<?php
// main.php

require_once 'config.php';

function getAccessToken()
{
  $credentials = base64_encode(CLIENT_ID . ':' . CLIENT_SECRET);

  $url = PAYPAL_API_BASE . '/v1/oauth2/token';
  $data = 'grant_type=client_credentials';

  $headers = [
    'Accept: application/json',
    'Authorization: Basic ' . $credentials,
    'Content-Type: application/x-www-form-urlencoded',
  ];

  $ch = curl_init($url);
  curl_setopt($ch, CURLOPT_POST, true);
  curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
  curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

  $response = curl_exec($ch);

  if (curl_errno($ch)) {
    // Handle the error case here
    curl_close($ch);
    return null;
  }

  curl_close($ch);

  $data = json_decode($response, true);
  return $data;
}

// Example usage:
$accessTokenData = getAccessToken();
if ($accessTokenData) {
  $accessToken = $accessTokenData['access_token'];
  echo "Access Token: " . $accessToken;
} else {
  echo "Failed to get the access token.";
}